### PR TITLE
chore: update one-shot Dockerfile to not depend on rust:ci

### PR DESCRIPTION
**Pull request by jacobmarble**
Related to #3119 and #3122

`docker build .` should just work, but it does not. The existing Dockerfile depends on image `quay.io/influxdb/rust:ci`, which is not published for ARM64, and doesn't seem to speed up the build process much anyhow.

This change simplifies that Dockerfile, though build times may increase.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Comment by crepererum** [comment]
This is NOT the dockerfile that we use for our prod images. We use https://github.com/influxdata/influxdb_iox/blob/main/docker/Dockerfile.iox . I'm wondering if we really need to dockerfiles here.

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,13 @@
 #syntax=docker/dockerfile:1.2
-FROM quay.io/influxdb/rust:ci as build
+FROM rust:1.56-slim-buster as build
```
Comment (e-dard) [inline Dockerfile:2]: Would `rust:slim-buster` use the latest stable release?

Comment (crepererum) [reply Dockerfile:2]: Please use `bullseye` instead of `buster` to be in line with our CI.

**Comment by jacobmarble** [comment]
> This is NOT the dockerfile that we use for our prod images. We use https://github.com/influxdata/influxdb_iox/blob/main/docker/Dockerfile.iox . I'm wondering if we really need to dockerfiles here.

I added this Dockerfile because I just want to `docker build .` Yes, I'm aware of docker/Dockerfile.iox, but it doesn't work on MacOS because it copies the binary from the local workstation (non-Linux) into a container image (Linux).

**Review by crepererum — APPROVED** [review]

**Pull request merged**